src/ast/passes/codegen_llvm.cpp

```diff
@@ -2948,10 +2960,7 @@ std::tuple<Value *, CodegenLLVM::ScopedExprDeleter> CodegenLLVM::getMapKey(
 AllocaInst *CodegenLLVM::getMultiMapKey(Map &map,
                                         const std::vector<Value *> &extra_keys)
 {
-  size_t size = 0;
-  for (auto &key_type : map.key_type.args_) {
-    size += key_type.GetSize();
-  }
+  size_t size = map.key_expr->type.GetSize();
```

Comment:
This previously read from `map.key_type` (the computed size of the map key across all AST nodes) but now uses `map.key_expr` (the size of a specific AST node).

I haven't triggered any obviously bad behaviour, but this script shows some bad generated code which I imagine could lead to issues with a bit more fiddling:
```
BEGIN { @[1,"aaaaaaaaaaaaaaaaaaaaaaaaaaa"] = hist(0); @[1,"a"] = hist(0);}
```
Different key sizes on each access:
```
  %"@_key2" = alloca [24 x i8], align 1
  %"@_key" = alloca [48 x i8], align 1
```
Out of bounds indexing:
```
  %14 = getelementptr [24 x i8], ptr %"@_key2", i64 0, i64 40
```